server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/StorageLinksService.scala

```diff
@@ -35,10 +40,27 @@ class StorageLinksService(storageLinks: StorageLinksCache) extends Http4sDsl[IO]
   //note: first param in the modify is the thing to do, second param is the value to return
   def createStorageLink(storageLink: StorageLink): IO[StorageLink] =
     storageLinks.modify { links =>
+
       val toAdd = List(storageLink.localBaseDirectory.path -> storageLink, storageLink.localSafeModeBaseDirectory.path -> storageLink).toMap
+
+      initializeDirectories(storageLink)
```

Comment:
you don't want to stick the side effect inside modify storagelinks. Instead chain the effect after modify
something like
```
for {
_ <- storageLinks.modify { links =>xxx}
_ <- initializeDirectories(storageLink)
} yield xxx